framework/mongo/__init__.py

```diff
@@ -31,6 +31,31 @@ def get_cache_key():
             return dummy_request
 
 
+def get_request_and_user_id():
```

Comment:
To CR: this was added because `get_cache_key` only returned the `request`, but a `user_id` was also required. Extending that to return both would break the `with_proxies` decorator on `StoredObject`. It should perhaps go elsewhere.